### DISCUSSION TITLE
1-1-relations with optional on both sides causing error

### DISCUSSION THREAD

**Discussion by eddeee888** [prisma/prisma#4462] [Q&A] [answered]
Hello,

I'm using Prisma 2.12.1 and getting errors when trying to have optional on both sides of a 1-1 relationship. More specifically, this is what I'm trying to do:

```prisma
model User {
  ...
  userZoomAccount      UserZoomAccount?    @relation(fields: [userZoomAccountId], references: [id])
  userZoomAccountId   Int?
}

model UserZoomAccount {
  ...
  user   User?
}
```

In this case, a user may or may not have a linked Zoom account. On the other hand, a Zoom account could be created without being linked to a user. I think the foreign key could be on either side in this case. I just chose to put it in `User`. 

When I'm trying to run the migration, I'm getting the following:
```
Error: The migration produced an invalid schema.
Datamodel error in schema that could not be rendered. Error parsing attribute "@relation": The relation field `User` on Model `UserZoomAccount` is required. This is invalid as it is not possible to enforce this constraint at the database level. Please make it optional instead.
```

Is this expected? I feel like in this case, it shouldn't error but keen to hear what everyone thinks :)

**Comment by ryands17** [comment, marked as answer]
Hey @eddeee888 👋 

I tried your schema with on a clean DB (without prior migrations) and it works fine after running the Migrate commands. Here's my schema:

```prisma
model User {
  id                Int              @id @default(autoincrement())
  userZoomAccount   UserZoomAccount? @relation(fields: [userZoomAccountId], references: [id])
  userZoomAccountId Int?
}

model UserZoomAccount {
  id   Int   @id @default(autoincrement())
  user User?
}
```

Could you try deleting your Migrations and check again? It might be related to a [change](https://github.com/prisma/prisma/releases/tag/2.12.0) made in version 2.12.0 regarding optional fields for 1-1 relations

**Reply by eddeee888** [reply]
Thanks @ryands17 , that worked! 

So when I delete my migrations, I'll have to remove all the records in `_Migration` table in the database right? 
Otherwise, I'll get the following error: `There are more migrations in the database than locally. This must not happen.`

If I do this in production, do I lose existing data when I rerun the migration command?

**Reply by ryands17** [reply]
@eddeee888 Yes you would need to remove the `_Migration` table as well. This should only remove the migration history and not any production data. I would still suggest running this in dev first to confirm :)

**Reply by eddeee888** [reply]
Thanks for confirming that :) I tested and it works as you mentioned. Thanks!

**Comment by AbdelrahmanAbounida** [comment]
change `userZoomAccountId   ` to 
`userZoomAccountId   Int? @unique`

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
